install/helm/agones/templates/crds/_gameserverspecschema.yaml

```diff
@@ -153,4 +153,10 @@ properties:
             type: integer
             title: The initial player capacity of this Game Server
             minimum: 0
+      immutableReplicas:
```

Comment:
I probably spent the longest time bikeshedding over the name of this. Other options I considered:

* `replicas` - no reason we can't just call it replicas and still disallow changes.
* `one` or `exactlyOne` - calling the field `one` or similar was an amusing possibility because it would make it obvious on error (e.g. `spec.one in body should be greater than or equal to 1`).

I settled here, but am open to other names.

Reply:
It's past 10pm here 😄 and I've had no sleep all week, so while I'm out of ideas for new names - I think I'd strongly recommend we implement this CRD changes behind a feature flag, and only add it if the flag is set.

Then we can also iterate on naming, or change our approach down the line without having a fear of making a breaking change (or breaking a user's experience).

Reply:
What if we just group it under the feature flag for #2974? Or do you think it would be better to have finer-granularity feature gates?

Reply:
No issue on my end to group it all together - it does seem like this all works together as a unit of work, unless you think otherwise?

Reply:
Nah, works great for me. Dumb question but I assume even the name of an alpha feature gate can be changed? :)

Reply:
I mean sure -- but it probably doesn't really matter what it's called, as long as it's vaguely semantic.

Reply:
Implemented under `LifecycleContract`. Did not publish the feature gate anywhere yet, but it means we can implement a lot of #2794 while talking it through.